test/unit/user_test.rb

```diff
@@ -617,27 +613,34 @@ def setup_user(operation)
 
   context 'auto create users' do
     setup do
-      ldap_attrs = { :firstname => "Foo", :lastname => "Bar", :mail => "baz@qux.com" }
-      AuthSourceLdap.any_instance.stubs(:authenticate).
-                                  returns(ldap_attrs)
+      ldap_attrs = { :firstname => "Foo", :lastname => "Bar", :mail => "baz@qux.com",
+                     :login => 'FoOBaR' }
+      AuthSourceLdap.any_instance.stubs(:authenticate).returns(ldap_attrs)
       @ldap_server = AuthSource.find_by_name("ldap-server")
     end
 
     test "enabled on-the-fly registration" do
       AuthSourceLdap.any_instance.expects(:update_usergroups).
-                                  with('fakeuser').returns(true)
+                                  with('FoOBaR').returns(true)
```

Comment:
This `FoOBaR` and the one in line 640 test that we're calling update_usergroups on `try_to_auto_create`.